packages/chain-bridge/src/router/BridgeRouter.sol

```diff
@@ -906,6 +906,79 @@ contract BridgeRouter is
         }
     }
 
+    /// @notice Governor-only force retry for a failed delivery with full override of adapter, type, and payload
+    /// @param failedOperationId The failed operation identifier to reference/clear on success
+    /// @param operationType The operation type to process
+    /// @param adapter The adapter address to attribute for peer checks and authorization
+    /// @param operationPayload The fully-formed payload to deliver
+    function forceRetryFailedDelivery(
+        bytes32 failedOperationId,
+        BridgeTypes.OperationType operationType,
+        address adapter,
+        bytes calldata operationPayload
+    ) external nonReentrant onlyGovernor whenNotPaused {
+        FailedDeliveryRecord memory r = failedDeliveries[failedOperationId];
+        if (r.failedAt == 0) revert FailureRecordNotFound();
+
```

Comment:
on line 869 there `r.failedAt == 0` reverts with `InvalidParams()` not `FailureRecordNotFound()` - it could also use the `getFailedDelivery()`